tools/build-tools/src/typeValidator/packageJson.ts

```diff
@@ -41,14 +44,39 @@ function createSortedObject<T>(obj:Record<string,T>): Record<string,T>{
     return sortedDeps;
 }
 
-export function getPackageDetails(packageDir: string): PackageDetails {
+function safeParse(json: string, error: string){
+    try{
+        return JSON.parse(json);
+    }catch{
+        throw new Error(error);
+    }
+}
+
+export async function getPackageDetailsOrThrow(packageDir: string, updateOptions?: {cwd?: string}):  Promise<PackageDetails> {
+    const result = await getPackageDetails(packageDir, updateOptions);
+    if(result.error !== undefined){
```

Comment:
wouldn't throwing the error directly in getPackageDetails be simpler?  since you're only using the non-throwing version in one place here and that could be wrapped in the try/catch

Reply:
they are handled a bit differently in the main loop. throws will result in a false return and a failure exit code where `error` will not.  maybe not named well. could use a different name for the property

Reply:
renamed to `skipReason`

Reply:
not super important now, but if we integrate into automation, it could matter